### PR TITLE
Inline methods in generated codes

### PR DESCRIPTION
**Pull request by tangjiangling**
## Description

After we have introduced `BytecodeExpressions`, we can inline and remove
some of the static methods in `CompilerOperations`.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by hashhar** [comment]
@tangjiangling Can you please rebase?

**Comment by tangjiangling** [comment]
done

File: core/trino-main/src/main/java/io/trino/sql/gen/IfCodeGenerator.java [hunk 1]
```diff
@@ -50,9 +52,10 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext context)
         BytecodeBlock conditionBlock = new BytecodeBlock()
                 .append(context.generate(condition))
                 .comment("... and condition value was not null")
-                .append(wasNull)
-                .invokeStatic(CompilerOperations.class, "not", boolean.class, boolean.class)
-                .invokeStatic(CompilerOperations.class, "and", boolean.class, boolean.class, boolean.class)
+                .append(not(wasNull))
```
Comment (lukasz-stec) [inline core/trino-main/src/main/java/io/trino/sql/gen/IfCodeGenerator.java:55]: are there tests that cover code generated by this and other changes?
Are all cases covered (`condition`, `wasNull` = `true` and`false`)?

**Review by lukasz-stec — COMMENTED** [review]
lgtm % comment

File: core/trino-main/src/main/java/io/trino/sql/gen/IfCodeGenerator.java (hunk 1 above)
Comment (tangjiangling) [reply core/trino-main/src/main/java/io/trino/sql/gen/IfCodeGenerator.java:55]: Not sure now, I'll take a look when I have time (maybe this weekend)

**Comment by tangjiangling** [comment]
CI hit #14814

**Comment by tangjiangling** [comment]
Convert to draft and I will re-activate it when I have time.

**Comment by mosabua** [comment]
:wave: @tangjiangling  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by ebyhr** [comment]
Let me close as a stale PR. Please feel free to reopen if you want to continue.

**Pull request closed without merging**